services/src/main/scala/cromwell/services/metadata/impl/WriteMetadataActor.scala

```diff
@@ -55,18 +62,28 @@ class WriteMetadataActor(batchSize: Int, flushRate: FiniteDuration)
     case Event(FlushBatchToDb, HasData(e)) =>
       log.debug("Flushing {} metadata events to the DB", e.length)
       // blech
-      val events = e.toVector.collect({ case e: MetadataEvent => e })
-      addMetadataEvents(events) onComplete {
-        case Success(_) => self ! DbWriteComplete
+      val events: Vector[MetadataEvent] = e.toVector.collect({ case e: MetadataEvent => e})
```

Comment:
do the `e.toVector` once. Using vectors here due to measured performance issues so try to do everything as a vector and then convert at the end.